### PR TITLE
Add core API using JWT authentication and tests for it [SCI-1507]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-1507

### CODE REVIEW

File: app/controllers/api/v20170715/core_api_controller.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          sample['name'] = s.name
+          samples_json << sample
+        end
+        render json: samples_json, status: :ok
+      end
+
+      private
+
+      def team_json(tm)
+        team = {}
```
Comment (Ducz0r) [inline app/controllers/api/v20170715/core_api_controller.rb]: Here, you can simply use `team = tm.as_json(only: [:id, :name, :description])`. Then, if we really need to rename `id` to `team_id`, call `team['team_id'] = team.delete('id')`.

File: app/controllers/api/v20170715/core_api_controller.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        team['team_id'] = tm.id.to_s
+        team['name'] = tm.name
+        team['description'] = tm.description
+        team
+      end
+
+      def project_json(pr)
+        project = {}
+        project['project_id'] = pr.id.to_s
```
Comment (Ducz0r) [inline app/controllers/api/v20170715/core_api_controller.rb]: Same, here, you can simply use `project = pr.as_json(only: [:id, :name, :visibility, :archived])`. Then, if we really need to rename `id` to `project_id`, call `project['project_id'] = project.delete('id')`.

File: app/controllers/api/v20170715/core_api_controller.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        project['name'] = pr.name
+        project['visibility'] = pr.visibility
+        project['archived'] = pr.archived
+        project
+      end
+
+      def experiment_json(exp)
+        experiment = {}
+        experiment['experiment_id'] = exp.id.to_s
```
Comment (Ducz0r) [inline app/controllers/api/v20170715/core_api_controller.rb]: Same, here, you can simply use `experiment = exp.as_json(only: [:id, :name, :description, :archived])`. Then, if we really need to rename `id` to `experiment_id`, call `experiment['experiment_id'] = experiment.delete('id')`.

File: app/controllers/api/v20170715/core_api_controller.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        experiment['name'] = exp.name
+        experiment['description'] = exp.description
+        experiment['archived'] = exp.archived
+        experiment
+      end
+
+      def task_json(tk)
+        task = {}
+        task['task_id'] = tk.id.to_s
```
Comment (Ducz0r) [inline app/controllers/api/v20170715/core_api_controller.rb]: Same, here, you can simply use `task = tk.as_json(only: [:id, :name, :description, :archived])`. Then, if we really need to rename `id` to `task_id`, call `task['task_id'] = task.delete('id')`.

File: config/initializers/core_api.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Api.configure do |config|
+  config.core_api_sign_alg = ENV['API_SIGN_ALG'] if ENV['CORE_API_SIGN_ALG']
+  config.core_api_token_ttl = ENV['API_TOKEN_TTL'] if ENV['CORE_API_TOKEN_TTL']
+  config.core_api_token_iss = ENV['API_TOKEN_ISS'] if ENV['CORE_API_TOKEN_ISS']
+end
```
Comment (Ducz0r) [inline config/initializers/core_api.rb]: More of a question really: so if you set `CORE_API_TOKEN_TTL` env. var to true (?), you need to also set `API_TOKEN_TTL` env. var with the actual value? Or am I misinterpreting this wrong?

Comment (okriuchykhin) [inline config/initializers/core_api.rb]: Sorry, this is typo, should be CORE_API_ everywhere

Comment (Ducz0r) [inline config/initializers/core_api.rb]: Cool :+1: Yeah then it's fine.

**Comment by Ducz0r** [comment]
Aight, :passenger_ship: away!

**Pull request merged**